src/net/sf/mpxj/primavera/PrimaveraReader.java

```diff
@@ -416,29 +512,93 @@ private ProjectCalendar getResourceCalendar(Integer calendarID)
    }
 
    /**
+    * Process resource rates
+    *
+    * @param rows resource rate data
+    */
+   public void processResourceRates(List<Row> rows)
+   {
+      // Primavera defines resource cost tables by start dates so sort and define end by next
+      rows.sort(new Comparator<Row>()
+      {
+         @Override public int compare(Row r1, Row r2)
+         {
+            Integer id1 = r1.getInteger("rsrc_id");
+            Integer id2 = r2.getInteger("rsrc_id");
+            int cmp = NumberHelper.compare(id1, id2);
+            if (cmp != 0)
+               return cmp;
+
+            Date d1 = r1.getDate("start_date");
+            Date d2 = r2.getDate("start_date");
+            return DateHelper.compare(d1, d2);
+         }
+      });
+
+      for (int i = 0; i < rows.size(); ++i)
+      {
+         Row row = rows.get(i);
+
+         Integer resourceID = row.getInteger("rsrc_id");
+         Rate standardRate = new Rate(row.getDouble("cost_per_qty"), TimeUnit.HOURS);
+         TimeUnit standardRateFormat = TimeUnit.HOURS;
+         Rate overtimeRate = new Rate(0, TimeUnit.HOURS); // does this exist in Primavera?
+         TimeUnit overtimeRateFormat = TimeUnit.HOURS;
+         Double costPerUse = NumberHelper.getDouble(0.0);
+         Double maxUnits = NumberHelper.getDouble(NumberHelper.getDouble(row.getDouble("max_qty_per_hr")) * 100); // adjust to be % as in MS Project
+         Date startDate = row.getDate("start_date");
+         Date endDate = DateHelper.LAST_DATE;
+
+         if (i + 1 < rows.size())
+         {
+            Row nextRow = rows.get(i + 1);
+            Integer nextResourceID = nextRow.getInteger("rsrc_id");
+            if (resourceID == nextResourceID)
```

Comment:
I'm getting a warning here about a suspicious integer comparison. Perhaps this instead:
```
            int nextResourceID = nextRow.getInt("rsrc_id");
            if (resourceID.intValue() == nextResourceID)
```